### ISSUE TITLE
Cannot define functions inside face! object

### ISSUE THREAD

**Issue by qtxie** [red/red#3122]
### Expected behavior
```
>> test: make face! [
[        type: 'base
[        myfunc:    does [ 'nothing ]
[    ]

== make object! [
    type: 'base
    offset: none
    size: none
...
```
### Actual behavior
```
>> test: make face! [
[        type: 'base
[        myfunc:    does [ 'nothing ]
[    ]

*** Where: do
*** Stack: load
*** Script Error: old has no value
*** Where: series?
*** Stack: series?
```
### Steps to reproduce the problem
Input following code into red console.
```
test: make face! [
    type: 'base
    myfunc:    does [ 'nothing ]
]
```
### Red version and build date, operating system with version.
Red for Windows version 0.6.3 built 28-Nov-2017

**Comment by qtxie** [comment]
FYI, this commit 9178838f7569b6e3d28fedcf070f04ab7e6e4e33 causes the error.

**Comment by greggirwin** [comment]
@endo64 found a regression due to this fix it seems.

**Comment by endo64** [comment]
This error still occurs if `system/view/debug? is true`
I have fixed it in PR #3073
